src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -320,19 +340,184 @@ public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBo
                 createdInstance.Process = new ProcessState { CurrentTask = "FormFilling_1", IsComplete = false };
             }
 
+            Instance storedInstance;
+
             try
             {
-                Instance result = await _instanceRepository.Create(createdInstance);
-                return Ok(result);
+                storedInstance = await _instanceRepository.Create(createdInstance);
             }
             catch (Exception e)
             {
                 logger.LogError($"Unable to create {appId} instance for {ownerId} due to {e}");
                 return StatusCode(500, $"Unable to create {appId} instance for {ownerId} due to {e}");
             }
+            
+            if (MultipartRequestHelper.IsMultipartContentType(Request.ContentType))
+            {
+                storedInstance = ReadhMultipartAndStoreFilesToBlob(Request, storedInstance, appInfo.ElementTypes, creationTime, userName, out ActionResult errorResult);
+
+                if (errorResult != null)
+                {
+                    return errorResult;
+                }
+            }
+            
+            return Ok(storedInstance);
+        }
+
+        /// <summary>
+        /// Loop through multipart content and save file(s) to blob
+        /// </summary>
+        /// <param name="request">The Http Request</param>
+        /// <param name="storedInstance">The Instance object to connect stored files</param>
+        /// <param name="appInfoElementTypes">The element types from the application info</param>
+        /// <param name="creationTime">The DateTime varialbe on which the storedInstance was created</param>
+        /// <param name="userName">The username of the ClaimsPrincipal</param>
+        /// <param name="errorResult">The possible error message</param>
+        /// <returns></returns>
+        private Instance ReadhMultipartAndStoreFilesToBlob(HttpRequest request, Instance storedInstance, List<ElementType> appInfoElementTypes, DateTime creationTime, string userName, out ActionResult errorResult)
+        {
+            errorResult = null;
+
+            MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(Request.ContentType);
+            string boundary = MultipartRequestHelper.GetBoundary(mediaType, _defaultFormOptions.MultipartBoundaryLengthLimit);
+
+            MultipartReader reader = new MultipartReader(boundary, Request.Body);
+
+            // Execute ReadNextSectionAsync() twice, to expose the second part of the Multipart content,
+            // first part has already been handled in ReadInstanceTemplateFromBody(HttpRequest req).
+            MultipartSection section = reader.ReadNextSectionAsync().Result;
```

Comment:
What if no instance is present. Then the first form-data is ignored!?

Reply:
Fixed with a check on all sections, if they contain instance instead of attachment.